src/inventory.cc

```diff
@@ -4756,25 +4756,42 @@ static int _barter_attempt_transaction(Object* a1, Object* a2, Object* a3, Objec
     return 0;
 }
 
-static int _barter_get_quantity_moved_items(Object* item, int maxQuantity, bool fromPlayerToNpc) {
+static int _barter_get_quantity_moved_items(
+    Object* item,
+    int maxQuantity,
+    bool fromPlayer,
+    bool fromInventory
+) {
     if (maxQuantity <= 1) {
         return maxQuantity;
     }
 
-    int quantityToMove = -1;
+    int suggestedValue = 1;
     if (item->pid == PROTO_ID_MONEY && !gGameDialogSpeakerIsPartyMember) {
         // Calculate change money automatically
         int totalCostPlayer = objectGetCost(_ptable);
         int totalCostNpc = _barter_compute_value(gDude, _target_stack[0]);
-        quantityToMove = fromPlayerToNpc ? std::min(totalCostNpc - totalCostPlayer, maxQuantity):
-                                           std::min(totalCostPlayer - totalCostNpc, maxQuantity);
-    }
-    // If the item is not money or player wants to add extra money
-    // then open window to set quantity manually
-    if (quantityToMove <= 0) {
-        quantityToMove = inventoryQuantitySelect(INVENTORY_WINDOW_TYPE_MOVE_ITEMS, item, maxQuantity);
-    }
-    return quantityToMove;
+        int balance = totalCostPlayer - totalCostNpc;
+        bool balancePositive = true;
+        if (balance < 0) {
+            balancePositive = false;
+            balance = -balance;
+        }
+        // fromPlayer | fromInventory | balancePositive | suggestedVale
+        //     0      |        0      |        0        |   abs(balance)
+        //     0      |        0      |        1        |      1
+        //     0      |        1      |        0        |      1
+        //     0      |        1      |        1        |   balance
+        //     1      |        0      |        0        |      1
+        //     1      |        0      |        1        |   balance
+        //     1      |        1      |        0        |   abs(balance)
+        //     1      |        1      |        1        |      1
+        // if balance 0 then suggestedVale is 1
+        if (balance != 0 && !(fromPlayer ^ fromInventory ^ balancePositive)) {
+            suggestedValue = std::min(balance, maxQuantity);
+        }
```

Comment:
This logic seems over-complicated. How does it handle cases:
1. Tables are balanced and you're dragging caps away from:
  a. Dude's table.
  b. NPC's table.
2. Tables are not balanced.

I think to simplify condition a bit, you can calculate `balancePositive` value in relative terms (if we're dragging to/from player then we need to subtract NPC's table from player's table, otherwise we invert that). This way the final condition can be simplified: 
`balance != 0 && fromInventory != balancePositive`